### ISSUE TITLE
LiteNetLibSyncList syncing

### ISSUE THREAD

**Issue by astromediaonly** [insthync/LiteNetLibManager#9]
I have ``LiteNetLibSyncList<MyClass>`` on my player object and im spawning two separate players on scene and to one of them im adding few items to that synclist on server but i see those items synced to both players instead of one i have added them to (Operation method is called with add value) ... its possible its some mess on my part but it seems ok - is there any chance this ``LiteNetLibSyncList`` is mistakenly synced to both player objects?

If i set ``forOwnerOnly = true`` it seems better (im not getting on client update from different object's list anymore) but im getting weird results on server - if i ask inside netfunction on server directly about count on that list it says 0 but if i ask on my stored entity in dictionary it says 2 (which is ok) ... im uterly confused what is happening. (It seems to be using always list on first spawned object inside that netfunction)

TL;DR: I just want to ask if you had used somewhere  ``LiteNetLibSyncList<MyClass>`` on multiple networkspawned objects and can tell me its working properly and no sync confusion between objects is happening in your case.

I want to ensure there is no bug involved because im running out of options i can look what could went wrong on my part. ~~Syncfields seems to behave correctly.~~ (EDIT: i have rewriten it to string syncfield and it does same weird thing :/ ... i guess objectid is messed up for some reason but i could be wrong ... i dont know how to debug it properly :( )

**Comment by astromediaonly** [comment]
Never mind, it would probably require too much comunication to resolve. I will try to put few breakpoints on server to see if everything is stored and spawned correctly there and if i will not see anything indicating where problem is i will rewrite it completely in different way and hopefully it will resolve the issue. (Im tempted to go back to normal packet sending, there i have more control over what is happening but sync-props should have been so usefull here, im trying to give them chance still + i have feeling this issue is something badly setup on lower level and this is just consequence and true cause will give me nightmares again with something else in future :/)

You are doing great stuff but in moments like this it makes me question if i should use it. When it works its great but when its not because i did some misstep because lacking documentation, examples or so i have nobody to ask for direction and i feel completely lost and helpless. I do usually prety complex stuff and i have no problem with it but i swear networking hates me :(

**Comment by insthync** [comment]
Next week I will write documents that explains everything, I think it can help you :), And also can help other people who want to learn how to create high level networking system.

**Comment by astromediaonly** [comment]
Ok, i appretiate it, thanks for response ... its nice to know there is still somebody on the other side :). Hopefully it will give me some more insight. I had probably used something in not intended way and i dont see it because it looks intuitive to me or i just missed some step in setting it up - sadly those type of issues are very hard to find :/ ... i was so close to have working inventory with equiping and unequiping stuff and it dynamicaly changing properties on model before i found this mess and its very frustrating ... it gives me hedaches for more than 3 days and i cant for god sake figure out whats wrong because there is so much stuff happening when you have 2 clients and syncfield between them and server so sorry if im too salty.

I dont like to bother you more than necessary ... i just wish you had few ppl doing support but thats probably unrealistic :) 

PS: about docs - it would be nice to add how fill synclist from list/arrray at once if its possible (atm im adding items one by one in loop) - that was first i wanted to find when i started using it and was not able to ... to send only one update ... but it seems its sending only one update even when i add them separatedly so its either optimised somehow or i dont see something else

**Comment by astromediaonly** [comment]
Ooh i had finaly figured out where problem was .. i think ... As im not yet familiar with all the stuff about network programming design i had forgot to add on few places ``IsOwnerClient`` (i did not realised not every ``IsClient`` is ``IsOwnerClient`` at that moment) and it broke hawoc in my code.

Now i have only one small issue - it seems when i do on server:
```
MySyncField.Value = "Value";
CallNetFunction(NetFnMyFunction, FunctionReceivers.All, value);
```
It seems i get on client that function call before syncfield value change ... syncfield is setup to ``DeliveryMethod.ReliableOrdered`` and ``CallNetFunction`` seems to be set this way by design. If you can tell me if its bug or i have to set something it would be highly appretiated. Sorry for being so "chatty" and "active" but game development is my big passion :)

**Comment by insthync** [comment]
That's because, sync field will start syncing in next its update time, based on its `sendInterval` setting, but `net function` message will be sent immediately.

**Comment by astromediaonly** [comment]
Oh i see, im sorry, i did not payed that much attention to its options because i was convinced it synces on every change ... that changes some things ... one followup questions if i may: if i set ``sendInterval`` to 0 will it reliably sync every change immediately so it will be delivered before netfn call on next line? (im trying to figure out if sending is drived by engine update function so it will ignore call order or by change it self if time is > than send interval) I will probably return it to synclist (which i believe do not have this behavior by design) now when i know where that issue was but its good to know for later use anyway - im still getting used to all that stuff i need to take into consideration when designing network releated logic :) ... thx again for your time.
